relay-server/src/services/processor.rs

```diff
@@ -806,19 +803,43 @@ impl<Group> ProcessEnvelopeState<Group> {
     fn remove_event(&mut self) {
         self.event = Annotated::empty();
     }
+}
 
-    /// Function for on-off switches that filter specific item types (profiles, spans)
-    /// based on a feature flag.
-    ///
-    /// If the project config did not come from the upstream, we keep the items.
-    fn should_filter(&self, feature: Feature, project_info: &ProjectInfo) -> bool {
-        match self.config.relay_mode() {
-            RelayMode::Proxy | RelayMode::Static | RelayMode::Capture => false,
-            RelayMode::Managed => !project_info.has_feature(feature),
-        }
+/// Function for on-off switches that filter specific item types (profiles, spans)
+/// based on a feature flag.
+///
+/// If the project config did not come from the upstream, we keep the items.
+fn should_filter(config: &Config, feature: Feature, project_info: &ProjectInfo) -> bool {
```

Comment:
Arguably you could just pass the mode in here, since that's the only relevant part of the config. This extends up to the callers of this function.

Reply:
nit: The argument list is now the feature sandwiched by two configs, I would either choose `(config, project_info, feature)` or `(feature, config, project_info)`.

Reply:
Will do! I did the same for other functions, thanks for the nit!